### PR TITLE
chore: bump alloy-eip7702

### PR DESCRIPTION
**Pull request by klkvr**
- Bump alloy, alloy-eip7702 and revm
- Fix Compact impl for SignedAuthorization
- Earlier we were using `Encodable`/`Decodable` impls  for encoding/decoding sidecars. Format for those was changed in https://github.com/alloy-rs/alloy/pull/1528 so I've replaced all `encode` with `rlp_encode_fields` and `decode` with `rlp_decode_fields`.

### CODE REVIEW

**Comment by klkvr** [comment]
book workspace is failing to compile (because main is broken after alloy bump)

it should get fine once this is merged though it's blocking some of the tests

**Review by shekhirin — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]
lgtm, blocked by revm release tho

**Review by onbjerg — APPROVED** [review]
lgtm, waiting on revm release

**Review by mattsse — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -411,9 +411,9 @@ reth-trie-db = { path = "crates/trie/db" }
 reth-trie-parallel = { path = "crates/trie/parallel" }
 
 # revm
-revm = { version = "16.0.0", features = ["std"], default-features = false }
+revm = { version = "17.0.0", features = ["std"], default-features = false }
 revm-inspectors = "0.9.0"
```
Comment (mattsse) [inline Cargo.toml]:
```suggestion
revm-inspectors = "0.10.0"
```

**Pull request merged**